### PR TITLE
Improve audio support

### PR DESCRIPTION
**Pull request by jlaako**
In order to support adjusting audio settings and to store/restore ALSA
audio device state, add alsa-utils and alsa-state packages.

Signed-off-by: Jussi Laako jussi.laako@linux.intel.com

### CODE REVIEW

**Comment by jlaako** [comment]
Added a patch to enable FLAC codec in gstreamer too. Since the library is being pulled in by the earlier change, we can as well enable the FLAC plugins in Gst.

**Comment by jlaako** [comment]
@juimonen this may be of interest to you.

**Comment by mythi** [comment]
@jlaako can you rebase this please

**Comment by jlaako** [comment]
@mythi done

**Comment by okartau** [comment]
test this please

File: meta-ostro-xt/classes/ostro-xt-image.bbclass [hunk 1]
```diff
@@ -70,3 +70,4 @@ IMAGE_INSTALL += "joule-udev-rules"
 
 # alsa-utils should support udev
 PACKAGECONFIG_append_pn-alsa-utils = " udev"
+PACKAGECONFIG_append_pn-gstreamer1.0-plugins-good = " flac"
```
Comment (mythi) [inline meta-ostro-xt/classes/ostro-xt-image.bbclass]: @jlaako Looks like 'flac' is in default PACKAGECONFIG. Can we drop this?

File: meta-ostro-xt/classes/ostro-xt-image.bbclass [hunk 2]
```diff
@@ -65,3 +67,6 @@ IMAGE_INSTALL += "${IMAGE_INSTALL_THERMAL_DAEMON}"
 
 # add package of Joule hardware specific udev rules
 IMAGE_INSTALL += "joule-udev-rules"
+
+# alsa-utils should support udev
+PACKAGECONFIG_append_pn-alsa-utils = " udev"
```
Comment (mythi) [inline meta-ostro-xt/classes/ostro-xt-image.bbclass]: looks like udev is default too. It should be safe to drop both PACKAGECONFIG_append lines

**Comment by jlaako** [comment]
@mythi how much we can trust on those defaults being set and not change with some update?

**Comment by mythi** [comment]
On Thu, Dec 8, 2016 at 12:06 PM, Jussi Laako wrote:

> @mythi <https://github.com/mythi> how much we can trust on those defaults
> being set and not change with some update?
>

100% trust. :) As long as there are no ostro-os updates that change those
made in this repo we're good.

**Comment by jlaako** [comment]
Updated, hopefully nobody never updates the base ostro.

**Comment by mythi** [comment]
These are now integrated.

**Pull request closed without merging**